tokio/tests/macros_select.rs

```diff
@@ -655,22 +655,40 @@ mod unstable {
     #[cfg(all(feature = "rt-multi-thread", not(target_os = "wasi")))]
     fn deterministic_select_multi_thread() {
         let seed = b"bytes used to generate seed";
+        let (tx, rx) = std::sync::mpsc::channel();
         let rt1 = tokio::runtime::Builder::new_multi_thread()
             .worker_threads(1)
+            .on_thread_park(move || tx.send(()).unwrap())
             .rng_seed(RngSeed::from_bytes(seed))
             .build()
             .unwrap();
+
+        // This makes sure that `enter_runtime()` from worker thread is called before the one from main thread,
+        // ensuring that the RNG state is consistent.
```

Comment:
There are two place to consume the `rng_seed` for multi thread runtime.

1. `block_on`
2. Launching worker threads.

Once the `on_thread_park` called, it is very likely that all worker_thread have started running, so this change reduces flakiness.

I leave more context here to help future developers debugging this test, it would be better to add this into the code, but this is not a blocker.